fecgov/fec-cms
# Feature/election page legacy tab redirects

## Summary
Redirect election page links that point to the old tab names to the corresponding new named-anchor link
Fix the issue with the page not jumping to the correct location for named-anchors regardless of wether it is a via one of these redirects or just a visit to a direct link with anchor suffix.

The old links that are affected include:
/?tab=totals
/?tab=contributions
/?tab=spending-by-others

- Resolves #2188

## Impacted areas of the application
modified:   data/urls.py
modified:   data/views.py
modified: data/templates/elections.jinja

- [x] The named anchor links used in the new election page structure do not consistently scroll the page to the correct position because the browser jumps to the correct position on-load and then loads additional assets which pushes  desired location to a random place in page.

## How to test:
- Checkout and run this branch locally: `feature/election-page-redirects`
- Go to the `election pages` and confirm that the page redirects from its legacy  `tab=` argument to the correct  `#` anchor suffix. Here is the mapping of tab to anchor:
  - 'contributions':'#individual-contributions',
  - 'totals' : '#candidate-financial-totals',
  - 'spending-by-others':'#independent-expenditures'
- Also confirm that the page jumps to the correct location on the page based on the named-anchor tag.

Test pages:
http://127.0.0.1:8000/data/elections/house/AL/1/2012/?tab=spending-by-others
http://127.0.0.1:8000/data/elections/house/AL/1/2012/?tab=contributions
http://127.0.0.1:8000/data/elections/house/AL/1/2012/?tab=totals
http://127.0.0.1:8000/data/elections/house/AL/2012/?tab=contributions (no district included)

http://127.0.0.1:8000/data/elections/president/2012/?tab=contributions
http://127.0.0.1:8000/data/elections/president/2012/?tab=spending-by-others
http://127.0.0.1:8000/data/elections/president/2012/?tab=totals
